packages/speech/src/index.js

```diff
@@ -0,0 +1,919 @@
+ *     Default: `1`.
+ * @param {boolean=} recognitionConfig.profanityFilter - See
+ *     [`RecognitionConfig.profanityFilter`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `false`.
+ * @param {string[]=} recognitionConfig.speechContext - See
+ *     [`RecognitionConfig.speechContext`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#google.cloud.speech.v1beta1.RecognitionConfig.FIELDS).
+ *     Default: `[]`.
+ * @param {function} callback - The callback function.
+ * @param {?error} callback.err - An error returned while making this request.
+ * @param {object[]} callback.syncRecognizeResponse -
+ *     `SyncRecognizeResponse` objects. See
+ *     [`SyncRecognizeResponse`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#syncrecognizeresponse).
+ * @param {object} callback.apiResponse - Raw API response. See
+ *     [`SyncRecognizeResponse`](https://cloud.google.com/speech/reference/rpc/google.cloud.speech.v1beta1#syncrecognizeresponse).
+ *
+ * @example
+ * //-
+ * // <h3>Run speech detection over a local file</h3>
+ * //-
+ * speech.detect('./bridge.raw', {
+ *   encoding: 'LINEAR16',
+ *   sampleRate: 16000
+ * }, function(err, result, apiResponse) {
+ *   //   result = {
+ *   //     results: [
+ *   //       {
+ *   //         alternatives: [
+ *   //           {
+ *   //             transcript: "...",
+ *   //             confidence: 0.88
+ *   //           }
+ *   //         ]
+ *   //       }
+ *   //     ]
+ *   //   }
+ * });
+ *
+ * //-
+ * // <h3>Run speech detection over a remote file</h3>
+ * //
+ * // *Note: This is not an officially supported feature of the Speech API. Our
+ * // library will make a request to the URL given, and send that upstream.*
+ * //-
+ * var file = 'https://your-domain.com/files/bridge.raw';
+ * speech.detect(file, {
+ *   encoding: 'LINEAR16',
+ *   sampleRate: 16000
+ * }, function(err, result, apiResponse) {});
+ *
+ * //-
+ * // <h3>Run speech detection over a file in Cloud Storage</h3>
+ * //-
+ * var file = 'gs://your-bucket-name/bridge.raw';
+ * speech.detect(file, {
+ *   encoding: 'LINEAR16',
+ *   sampleRate: 16000
+ * }, function(err, result, apiResponse) {});
+ */
+Speech.prototype.detect = function(file, config, callback) {
+  var self = this;
+
+  Speech.findFiles_(file, function(err, files) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    var reqOpts = {
+      config: extend({
+        encoding: Speech.detectEncoding_(file)
+      }, config),
+      audio: files[0]
+    };
+
+    self.recognize(reqOpts, callback);
+  });
+};
+
+/**
+ * Perform asynchronous speech-recognition: receive results via the
+ * `google.longrunning.Operations` interface.
+
+ * This method sends audio to the Speech API, which immediately responds with a
+ * job ID. You can then poll the job for a result. This is useful if your app
```

Comment:
I think somewhere I read that you should use the async version when dealing with larger file sizes - does that sound right?

We've also received some feedback that the phrasing for this doesn't sound quite right. Suggested text was "this is useful if you're willing to do the work in the background and be notified when things are all done".